File: 01-Numbers/py/numbers.py
if __name__ == "__main__":
    # integers
    a = -42
    b = 56

    large_number = 123_456_789  # python >= 3.6

    # floats
    f = -1.2345
   
    # complex numbers
    complex_integer = complex(10, 20)
    complex_float = complex(10.0, 20.0000000000001)

    print(f"{complex_integer} == {complex_float}", complex_integer == complex_float)

    # binary literals
    b1 = 0b1010_1010
    b2 = 0b0101_0101
    assert b1 | b2 == 255
    assert b1 & b2 == 0 
    print(b1 | b2)
    print(b1 & b2)

    # hex literals
    hx = 0xCAFE_BABE
    print(hx)

    # boolean
    yes = True
    no = False

File: 01-Numbers/rs/src/numbers.rs
use num_complex::Complex;

#[allow(unused_variables)]
pub fn counting() {
    // integers
    let x: i32 = -1000;
    let x = -1000i32;
    let y: u64 = 100;
    let y = 100u64;
    let large_number = 100_000_000;

    // floats
    let xf: f32 = -1.2345;
    let xf = -1.2345_f32;

    // complex numbers
    // not part of rusts standard library
    let complex_integer = Complex::new(10, 20);
    let complex_float = Complex::new(10.1, 20.1);

    // binary literals
    let x: u8 = 0b1010_1010;
    let y: u8 = 0b0101_0101;
    dbg!(x | y);
    assert_eq!(x | y, 255);
    assert_eq!(x & y, 0);

    // hex
    let hx = 0xCAFE_BABE_u32;

    // boolean
    let yes = true;
    let no = false;
}
